scoring_engine/web/views/api/service.py

```diff
@@ -57,62 +57,72 @@ def service_get_checks(service_id):
 def update_service_account_info():
     if current_user.is_white_team or current_user.is_blue_team:
         if "name" in request.form and "value" in request.form and "pk" in request.form:
+            if request.form["name"] == "username":
+                modify_usernames_setting = Setting.get_setting("blue_team_update_account_usernames")
+                if modify_usernames_setting.value is False and current_user.is_blue_team:
+                    return jsonify({"error": "Incorrect permissions"})
+
+            elif request.form["name"] == "password":
+                modify_passwords_setting = Setting.get_setting("blue_team_update_account_passwords")
+                if modify_passwords_setting.value is False and current_user.is_blue_team:
+                    return jsonify({"error": "Incorrect permissions"})
+
             account = session.query(Account).get(int(request.form["pk"]))
-            if current_user.team == account.service.team or current_user.is_white_team:
-                if account:
+            if account:
+                if current_user.team == account.service.team or current_user.is_white_team:
```

Comment:
swapped order because need to check if "account" exists before we can access its properties